src/transformers/generation_beam_constraints.py

```diff
@@ -249,13 +422,20 @@ def advance(self):
         Though we don't care which constraint is fulfilled first, if we are in the progress of fulfilling a constraint,
         that's the only one we'll return.
         """
+        token_list = []
         if self.inprogress_constraint is None:
-            token_list = []
             for constraint in self.pending_constraints:  # "pending" == "unfulfilled yet"
                 advance = constraint.advance()
-                token_list.append(advance)
+                if len(advance.size()) > 0:
+                    token_list = token_list + [a for a in advance]
+                else:
+                    token_list.append(advance)
```

Comment:
If the tensor is not empty, you add `.extend(` token_list with a list of tensors,
If the tensor is empty, you add it to the list ? (ending up with `[...., torch.Tensor([])]` )

If that's true, that seems very wrong in terms of types.